lib/home_page.dart

```diff
@@ -8,15 +24,288 @@ class HomePage extends StatefulWidget {
+                ),
+                const SizedBox(
+                  height: 35,
+                )
+              ],
+            ),
+          ],
+        ),
       ),
     );
   }
+
+  Future<void> _loadNewRideEvent(int rideEventId) async {
+    final Map<String, dynamic> data = await SupabaseManager.supabaseClient.from('ride_events').select('''
+      *,
+      ride: ride_id(*,
+        rider: rider_id(*),
+        drive: drive_id(*,
+          driver: driver_id(*)
+          )
+      )
+      ''').eq('id', rideEventId).single();
+    final RideEvent rideEvent = RideEvent.fromJson(data);
+    if (rideEvent.isForCurrentUser()) {
+      setState(() {
+        _items.insert(0, rideEvent);
+      });
+    }
+  }
+
+  Future<void> _loadNewMessage(int messageId) async {
+    final Map<String, dynamic> data = await SupabaseManager.supabaseClient.from('messages').select('''
+      *,
+      sender: sender_id(*)
+      )
+    ''').eq('id', messageId).single();
+    final Message message = Message.fromJson(data);
+    if (message.read == false) {
+      setState(() {
+        _items.insert(0, message);
+      });
+    }
+  }
+
+  Card _buildMessageWidget(Message message, BuildContext context) {
+    return Card(
+      child: InkWell(
+        child: ListTile(
+          leading: Avatar(message.sender!),
+          title: Text(message.sender!.username),
+          subtitle: Text(message.content),
+          trailing: Text(
+            DateFormat('HH:mm').format(message.createdAt!),
+            style: Theme.of(context).textTheme.caption,
+          ),
+          onTap: () {
+            Navigator.of(context).push(
+              MaterialPageRoute<void>(
+                builder: (BuildContext context) => ChatPage(
+                  chatId: message.chatId,
+                  profile: message.sender!,
+                ),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+
+  Widget _buildRideEventWidget(RideEvent rideEvent, BuildContext context) {
+    final bool isforDrive = rideEvent.ride!.riderId != SupabaseManager.getCurrentProfile()!.id!;
+    return Card(
+      child: InkWell(
+        child: ListTile(
+          leading: isforDrive ? const Icon(Icons.drive_eta) : const Icon(Icons.chair),
+          title: Text(rideEvent.getTitle(context)),
+          subtitle: Text(rideEvent.getMessage(context)),
+          trailing: Text(
+            DateFormat('HH:mm').format(rideEvent.createdAt!),
+            style: Theme.of(context).textTheme.caption,
+          ),
+          onTap: () {
+            rideEvent.markAsRead();
+            Navigator.of(context).push(
+              MaterialPageRoute<void>(
+                builder: (BuildContext context) =>
+                    isforDrive ? DriveDetailPage(id: rideEvent.ride!.driveId) : RideDetailPage(id: rideEvent.rideId),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+
+  Widget _buildWidget(Model model, BuildContext context) {
```

Comment:
1) I changed the space to 1 between the cards is that okay or should i remove it completly?
2) Yes I thought that too, for the messages i don't think it's a big problem since it's clear for wich person the chat is. And for the events, i can add somekind of mini-timeline but i dont think that could replace the icon since it's way to big. What i could do is just add it in the subtitle?
3) yes i forgot that